server/player/player.go

```diff
@@ -940,7 +961,14 @@ func (p *Player) dropContents() {
 // Respawn spawns the player after it dies, so that its health is replenished and it is spawned in the world
 // again. Nothing will happen if the player does not have a session connected to it.
 func (p *Player) Respawn() {
-	w := p.World()
+	position, w, ok := p.realSpawnPos()
+	if bl, ok := w.Block(position).(block.SpawnBlock); ok {
+		bl.Update(position, p, w)
+	}
+	pos := position.Vec3Middle()
+	if ok {
+		p.Messaget("%tile.bed.notValid")
+	}
```

Comment:
Should this ok check not be moved up to under line 964 where it's originally defined? Should it not also return?

Reply:
it should not return otherwise the player would not respawn

Reply:
Ah yeah I misunderstood the code, the first haf of my message should still be acted on though

Reply:
already done